cpp/dolfinx/mesh/generation.h

```diff
@@ -333,43 +338,42 @@ Mesh<T> build_tet(MPI_Comm comm, MPI_Comm subcomm,
   common::Timer timer("Build BoxMesh (tetrahedra)");
   std::vector<T> x;
   std::vector<std::int64_t> cells;
-  if (subcomm != MPI_COMM_NULL)
-  {
-    x = create_geom<T>(subcomm, p, n);
+  fem::CoordinateElement<T> element(CellType::tetrahedron, 1);
 
-    const std::int64_t nx = n[0];
-    const std::int64_t ny = n[1];
-    const std::int64_t nz = n[2];
-    const std::int64_t n_cells = nx * ny * nz;
+  if (subcomm == MPI_COMM_NULL)
+    return create_mesh(comm, subcomm, cells, element, subcomm, x,
+                       {x.size() / 3, 3}, partitioner);
 
-    std::array range_c = dolfinx::MPI::local_range(
-        dolfinx::MPI::rank(subcomm), n_cells, dolfinx::MPI::size(subcomm));
-    cells.reserve(6 * (range_c[1] - range_c[0]) * 4);
+  x = create_geom<T>(subcomm, p, n);
```

Comment:
Note sure, the previous `create_mesh` call requires the empty `x` - and since its the template argument of `create_mesh` we would probably need quite extensive type hinting without it.

Reply:
Off the table with revert of the `if/else` treatment.